.github/actions/issue-labeler/src/index.ts

```diff
@@ -0,0 +1,107 @@
+import { setFailed, debug } from '@actions/core'
+import { context, getOctokit } from '@actions/github'
+
+type GitHubClient = ReturnType<typeof getOctokit>['rest']
+
+async function run() {
+  const token = process.env.GITHUB_TOKEN
+  if (!token) throw new Error('No GITHUB_TOKEN provided')
+
+  const { issue } = context.payload
+  if (!issue) return console.log('Not an issue, exiting')
+
+  const { body: issue_body, number: issue_number, title: issue_title } = issue
+  if (!issue_number) return console.log('Could not get issue number, exiting')
+  if (!issue_body) return console.log('Could not get issue body, exiting')
+  if (!issue_title) return console.log('Could not get issue title, exiting')
+
+  // A client to load data from GitHub
+  const { rest: client } = getOctokit(token)
+
+  // Load our regex rules from the repo labels
+  const labels = await loadAreaLabels(client)
+
+  debug(`Loaded labels: ${Array.from(labels.keys()).join(', ')}`)
+
+  /** List of labels to add */
+  const toAdd: string[] = []
+
+  // https://github.com/vercel/next.js/blame/canary/.github/ISSUE_TEMPLATE/1.bug_report.yml
+
+  const matchSection = issue_body
+    .split('Which area(s) of Next.js are affected? (leave empty if unsure)')[1]
+    ?.split('Link to the code that reproduces this issue')[0]
+
+  if (!matchSection) {
+    console.log(
+      `Issue #${issue_number} does not contain a match section, likely not a bug template, exiting`
+    )
+    return
+  }
+
+  debug(`Match section: ${matchSection}`)
+
+  for (const [label, description] of labels.entries()) {
+    if (matchSection.includes(description)) {
+      toAdd.push(label)
+    }
+  }
+
+  debug(`Labels to add: ${toAdd.join(', ')}`)
+
+  let promises = []
+  if (toAdd.length) promises.push(addLabels(client, issue_number, toAdd))
+
+  const rejected = (await Promise.allSettled(promises))
+    .map((p) => p.status === 'rejected' && p.reason)
+    .filter(Boolean)
+
+  if (rejected.length) throw new AggregateError(rejected)
```

Comment:
Do you need this? It's just one promise, right?

Reply:
Oh, good point! This is some leftover.